libvast/src/detail/posix.cpp

```diff
@@ -254,21 +255,23 @@ bool read(int fd, void* buffer, size_t bytes, size_t* got) {
   return true;
 }
 
-bool write(int fd, const void* buffer, size_t bytes, size_t* put) {
+caf::error write(int fd, const void* buffer, size_t bytes, size_t* put) {
   auto total = size_t{0};
   auto buf = reinterpret_cast<const uint8_t*>(buffer);
   while (total < bytes) {
     ssize_t written;
     do {
       written = ::write(fd, buf + total, bytes - total);
     } while (written < 0 && errno == EINTR);
-    if (written <= 0)
-      return false;
+    if (written < 0)
+      return make_error(ec::filesystem_error, "write(2)", std::strerror(errno));
+    if (written == 0)
+      return make_error(ec::filesystem_error, "write(2) returned 0");
```

Comment:
Could we add a comment here why returning 0 is a special error case?

Unrelated to this change, I wonder what one should do for 0? Try again? Is the combo of 0 and `EINTR` valid?

Reply:
According to the [man page](https://linux.die.net/man/2/write) `errno` may or may not be set if `write` returns 0. This return value should only be possible if `bytes` is 0, but in that case we don't enter the outer loop. The second check exists for totality, but I don't expect this to ever happen in practice.

Reply:
Your comment here makes sense, but the code doesn't check for `bytes == 0 && written == 0`.

Reply:
`written` is initialized to 0, so if the function is called with `bytes` set to 0 as well, we don't even enter the outer loop.